src/truncate/truncate.scss

```diff
@@ -29,3 +29,38 @@
     max-width: 10000px !important;
   }
 }
+
+.Truncate {
+  display: inline-flex;
+  min-width: 0;
+  max-width: 100%;
+
+  .Truncate-text {
+    min-width: 3em;
+    max-width: 100%;
+    max-width: fit-content;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+
+    + .Truncate-text {
+      margin-right: $spacer-1;
+      margin-left: $spacer-1;
+    }
+
+    &.Truncate-text--beginning {
+      direction: rtl;
```

Comment:
Is it safe to use `direction: rtl` for reasons other than identifying the writing system direction? I wonder if this could be confusing for any user-provided text that uses a `rtl` system.

Reply:
A good question that I was hoping to answer before shipping this. 😬

Reply:
I _vaguely_ remember trying something like this in the past but can't really remember the details. I'll see if I can find it again.

Reply:
Found it: https://github.com/github/github/pull/119636. Not sure if it's still an issue in Safari. The other problem was with dotfiles where the dot moved to the end:

![Screen Shot 2021-04-26 at 17 25 41](https://user-images.githubusercontent.com/378023/116052470-9b9cca00-a6b4-11eb-9bb7-201e4f821994.png)